### DISCUSSION TITLE
CIDR whitelists for automation tokens

### DISCUSSION THREAD

**Discussion by deppy** [npm/feedback#369] [Registry] [answered]
Automation tokens were recently added but the [CIDR whitelist](https://docs.npmjs.com/using-private-packages-in-a-ci-cd-workflow#cidr-whitelists) functionality wasn't expanded to support the automation tokens. We would like to see the CIDR whitelist functionality expanded to also support the newly added automation token type.

**Comment by MylesBorins** [comment, marked as answer]
Hey @deppy thanks for opening this discussion. Currently we only support creating a token with a CIDR allow list when creating the token from the CLI, but you currently cannot create automation tokens from the CLI.

Seems like there are in turn 3 potentially different things we would need to do in order to support this

1. support creating automation tokens from the npm CLI
2. support adding CIDR allow lists from the npmjs.com UI
3. support CIDR allow lists for automation tokens

I'll need to dig into each of these independently with a couple different teams to figure out what we can currently support and how much work it would be to add each of these individual features.

**Reply by MylesBorins** [reply]
@deppy this is something we have on the roadmap but we do need to put it on the backlog for right now. I've kept a reference to this discussion and will update you if priorities change

**Reply by AaronSterlingGENEICD** [reply]
Requesting an update on this. We'd like the feature too.

**Reply by MylesBorins** [reply]
@AaronSterlingGENEICD this is still on the radar but not currently prioritized for the website. The CLI team have a ticket on their backlog and it will hopefully get picked up in the near future

**Comment by maxpain** [comment]
any updates?

**Comment by qgustavor** [comment]
Can whitelists support, along with CIDR address ranges, support for services such as GitHub Actions, Cloudflare and other popular services that already provide lists (sometimes huge lists) of CIDR address ranges? You see, GitHub Actions use 3325 address ranges (list available [here](https://api.github.com/meta)). Even if determining what counts as "popular services" is too hard, at least GitHub Actions could be considered as it's really hard to set up a token to work only for Actions. Also, surely, if a token locked to Actions get leaked someone could just GH Actions to use it, but it is still useful considering security in depth.
